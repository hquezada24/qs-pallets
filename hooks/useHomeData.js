import { useState, useEffect } from "react";

export const useHomeData = () => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  const API_BASE_URL =
    import.meta.env.VITE_API_BASE_URL || "http://localhost:3000";

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await fetch(API_BASE_URL);
        if (!response.ok) throw new Error("Failed to fetch");
        const result = await response.json();
        setData(result);
      } catch (err) {
        setError(err.message);
      } finally {
        setLoading(false);
      }
    };

    fetchData();
  }, []);

  return { data, loading, error };
};
