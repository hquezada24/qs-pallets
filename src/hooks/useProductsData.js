import { useState, useEffect } from "react";

export const useProductsData = () => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const API_BASE_URL =
      import.meta.env.VITE_API_PRODUCTS || "http://localhost:3000/products";
    const fetchData = async () => {
      try {
        const response = await fetch(API_BASE_URL);
        if (!response.ok) throw new Error("Failed to fetch");
        const result = await response.json();
        setData(result);
      } catch (err) {
        setError(err.message);
      } finally {
        setLoading(false);
      }
    };

    fetchData();
  }, []);

  return { data, loading, error };
};
