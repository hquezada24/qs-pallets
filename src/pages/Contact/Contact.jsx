import { useState } from "react";
import { ScrollRestoration } from "react-router-dom";
import { Button } from "../../components/common/Button/Index";
import { useContactData } from "../../../hooks/useContactData";
import styles from "./Styles.module.css";

const Contact = () => {
  const [formData, setFormData] = useState({
    fullName: "",
    email: "",
    phone: "",
    companyName: "",
    subject: "",
    message: "",
  });
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [submitStatus, setSubmitStatus] = useState(null);
  const [errors, setErrors] = useState({});
  const API_BASE_URL =
    import.meta.env.VITE_API_BASE_URL || "http://localhost:3000";
  const { data, loading, error } = useContactData();

  const validateForm = () => {
    const newErrors = {};

    if (!formData.fullName.trim()) {
      newErrors.fullName = "Name is required";
    }

    if (!formData.email.trim()) {
      newErrors.email = "Email is required";
    } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) {
      newErrors.email = "Please enter a valid email address";
    }

    if (!formData.phone.trim()) {
      newErrors.phone = "Phone number is required";
    } else if (
      !/^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/.test(formData.phone)
    ) {
      newErrors.phone = "Please enter a valid phone number";
    }

    if (!formData.subject.trim()) {
      newErrors.subject = "Subject is required";
    }

    if (!formData.message.trim()) {
      newErrors.message = "Message is required";
    } else if (formData.message.trim().length < 10) {
      newErrors.message = "Message must be at least 10 characters long";
    }

    setErrors(newErrors);
    return Object.keys(newErrors).length === 0;
  };

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setFormData((prev) => ({ ...prev, [name]: value }));

    // Clear error for this field when user starts typing
    if (errors[name]) {
      setErrors((prev) => ({ ...prev, [name]: "" }));
    }
  };

  const handleSubmit = async (e) => {
    e.preventDefault();

    console.log(formData);

    if (!validateForm()) {
      return;
    }

    setIsSubmitting(true);
    setSubmitStatus(null);

    try {
      // Make the API call
      const response = await fetch(`${API_BASE_URL}/contact`, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(formData), // Send the actual formData
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      // send the data to backend
      console.log("Form submitted:", formData);

      setSubmitStatus("success");
      setFormData({
        fullName: "",
        email: "",
        phone: "",
        companyName: "",
        subject: "",
        message: "",
      });
    } catch (error) {
      setSubmitStatus("error");
      console.error("Form submission error:", error);
    } finally {
      setIsSubmitting(false);
    }
  };

  if (loading) return <div className={styles.loading}>Loading...</div>;
  if (error) return <div className={styles.error}>Error: {error}</div>;

  return (
    <div className={styles.contactPage}>
      <ScrollRestoration />
      <div className={styles.hero}>
        <div className={styles.heroContent}>
          <h1 className={styles.title}>Get In Touch</h1>
          <p className={styles.subtitle}>
            Ready to discuss your pallet needs? We're here to help with custom
            solutions and reliable supply.
          </p>
        </div>
      </div>

      <div className={styles.container}>
        <div className={styles.content}>
          {/* Contact Form */}
          <div className={styles.formSection}>
            <div className={styles.formHeader}>
              <h2>Send Us a Message</h2>
              <p>
                Fill out the form below and we'll get back to you within 24
                hours.
              </p>
            </div>

            {submitStatus === "success" && (
              <div className={styles.successMessage} role="alert">
                <svg
                  className={styles.successIcon}
                  viewBox="0 0 24 24"
                  fill="currentColor"
                >
                  <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z" />
                </svg>
                <span>
                  Thank you! Your message has been sent successfully. We'll
                  contact you soon.
                </span>
              </div>
            )}

            {submitStatus === "error" && (
              <div className={styles.errorMessage} role="alert">
                <svg
                  className={styles.errorIcon}
                  viewBox="0 0 24 24"
                  fill="currentColor"
                >
                  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z" />
                </svg>
                <span>
                  Sorry, there was an error sending your message. Please try
                  again or contact us directly.
                </span>
              </div>
            )}

            <form className={styles.form} onSubmit={handleSubmit} noValidate>
              <div className={styles.formGroup}>
                <label htmlFor="name" className={styles.label}>
                  Full Name *
                </label>
                <input
                  type="text"
                  id="name"
                  name="fullName"
                  value={formData.fullName}
                  onChange={handleInputChange}
                  className={`${styles.input} ${
                    errors.name ? styles.inputError : ""
                  }`}
                  placeholder="Enter your full name"
                  required
                  aria-describedby={errors.name ? "name-error" : undefined}
                />
                {errors.name && (
                  <span
                    id="name-error"
                    className={styles.errorText}
                    role="alert"
                  >
                    {errors.name}
                  </span>
                )}
              </div>

              <div className={styles.formRow}>
                <div className={styles.formGroup}>
                  <label htmlFor="email" className={styles.label}>
                    Email Address *
                  </label>
                  <input
                    type="email"
                    id="email"
                    name="email"
                    value={formData.email}
                    onChange={handleInputChange}
                    className={`${styles.input} ${
                      errors.email ? styles.inputError : ""
                    }`}
                    placeholder="your.email@company.com"
                    required
                    aria-describedby={errors.email ? "email-error" : undefined}
                  />
                  {errors.email && (
                    <span
                      id="email-error"
                      className={styles.errorText}
                      role="alert"
                    >
                      {errors.email}
                    </span>
                  )}
                </div>

                <div className={styles.formGroup}>
                  <label htmlFor="phone" className={styles.label}>
                    Phone Number *
                  </label>
                  <input
                    type="tel"
                    id="phone"
                    name="phone"
                    value={formData.phone}
                    onChange={handleInputChange}
                    className={`${styles.input} ${
                      errors.phone ? styles.inputError : ""
                    }`}
                    placeholder="(000) 000-0000"
                    required
                    aria-describedby={errors.phone ? "phone-error" : undefined}
                  />
                  {errors.phone && (
                    <span
                      id="phone-error"
                      className={styles.errorText}
                      role="alert"
                    >
                      {errors.phone}
                    </span>
                  )}
                </div>
              </div>

              <div className={styles.formRow}>
                <div className={styles.formGroup}>
                  <label htmlFor="company" className={styles.label}>
                    Company Name
                  </label>
                  <input
                    type="text"
                    id="company"
                    name="companyName"
                    value={formData.companyName}
                    onChange={handleInputChange}
                    className={styles.input}
                    placeholder="Your Company"
                  />
                </div>
              </div>

              <div className={styles.formGroup}>
                <label htmlFor="subject" className={styles.label}>
                  Subject *
                </label>
                <input
                  type="text"
                  id="subject"
                  name="subject"
                  value={formData.subject}
                  onChange={handleInputChange}
                  className={`${styles.input} ${
                    errors.subject ? styles.inputError : ""
                  }`}
                  placeholder="Brief description of your inquiry"
                  required
                  aria-describedby={
                    errors.subject ? "subject-error" : undefined
                  }
                />
                {errors.subject && (
                  <span
                    id="subject-error"
                    className={styles.errorText}
                    role="alert"
                  >
                    {errors.subject}
                  </span>
                )}
              </div>

              <div className={styles.formGroup}>
                <label htmlFor="message" className={styles.label}>
                  Message *
                </label>
                <textarea
                  id="message"
                  name="message"
                  value={formData.message}
                  onChange={handleInputChange}
                  className={`${styles.textarea} ${
                    errors.message ? styles.inputError : ""
                  }`}
                  placeholder="Please provide details about your pallet requirements, quantities, specifications, or any questions you have."
                  rows="6"
                  required
                  aria-describedby={
                    errors.message ? "message-error" : undefined
                  }
                />
                {errors.message && (
                  <span
                    id="message-error"
                    className={styles.errorText}
                    role="alert"
                  >
                    {errors.message}
                  </span>
                )}
              </div>

              <div className={styles.formActions}>
                <Button
                  text={isSubmitting ? "Sending..." : "Send Message"}
                  type="submit"
                  variant="primary"
                  size="large"
                  loading={isSubmitting}
                  disabled={isSubmitting}
                  ariaLabel="Submit contact form"
                />
              </div>
            </form>
          </div>

          {/* Contact Information */}
          <div className={styles.infoSection}>
            <div className={styles.infoCard}>
              <h3 className={styles.infoTitle}>Contact Information</h3>

              <div className={styles.infoItem}>
                <div className={styles.infoIcon}>
                  <svg viewBox="0 0 24 24" fill="currentColor">
                    <path d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z" />
                  </svg>
                </div>
                <div className={styles.infoContent}>
                  <h4>Our Location</h4>
                  <p>{data.contactInfo.address.street}</p>
                  <p>
                    {data.contactInfo.address.city},{" "}
                    {data.contactInfo.address.state}{" "}
                    {data.contactInfo.address.zipCode}
                  </p>
                </div>
              </div>

              <div className={styles.infoItem}>
                <div className={styles.infoIcon}>
                  <svg viewBox="0 0 24 24" fill="currentColor">
                    <path d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z" />
                  </svg>
                </div>
                <div className={styles.infoContent}>
                  <h4>Phone</h4>
                  <p>
                    <a
                      href={data.contactInfo.phone.href}
                      className={styles.contactLink}
                    >
                      {data.contactInfo.phone.display}
                    </a>
                  </p>
                </div>
              </div>

              <div className={styles.infoItem}>
                <div className={styles.infoIcon}>
                  <svg viewBox="0 0 24 24" fill="currentColor">
                    <path d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z" />
                  </svg>
                </div>
                <div className={styles.infoContent}>
                  <h4>Email</h4>
                  <p>
                    <a
                      href={data.contactInfo.email.href}
                      className={styles.contactLink}
                    >
                      {data.contactInfo.email.display}
                    </a>
                  </p>
                </div>
              </div>

              <div className={styles.infoItem}>
                <div className={styles.infoIcon}>
                  <svg viewBox="0 0 24 24" fill="currentColor">
                    <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z" />
                    <path d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z" />
                  </svg>
                </div>
                <div className={styles.infoContent}>
                  <h4>Business Hours</h4>
                  <p>{data.contactInfo.businessHours.weekdays}</p>
                  <p>{data.contactInfo.businessHours.saturday}</p>
                  <p>{data.contactInfo.businessHours.sunday}</p>
                </div>
              </div>
            </div>

            <div className={styles.infoCard}>
              <h3 className={styles.infoTitle}>Why Choose QS Pallets?</h3>
              <ul className={styles.benefitsList}>
                <li className={styles.benefit}>
                  <svg
                    className={styles.checkIcon}
                    viewBox="0 0 24 24"
                    fill="currentColor"
                  >
                    <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z" />
                  </svg>
                  <span>Fast turnaround times</span>
                </li>
                <li className={styles.benefit}>
                  <svg
                    className={styles.checkIcon}
                    viewBox="0 0 24 24"
                    fill="currentColor"
                  >
                    <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z" />
                  </svg>
                  <span>Custom pallet solutions</span>
                </li>
                <li className={styles.benefit}>
                  <svg
                    className={styles.checkIcon}
                    viewBox="0 0 24 24"
                    fill="currentColor"
                  >
                    <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z" />
                  </svg>
                  <span>Competitive pricing</span>
                </li>
                <li className={styles.benefit}>
                  <svg
                    className={styles.checkIcon}
                    viewBox="0 0 24 24"
                    fill="currentColor"
                  >
                    <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z" />
                  </svg>
                  <span>Quality materials</span>
                </li>
                <li className={styles.benefit}>
                  <svg
                    className={styles.checkIcon}
                    viewBox="0 0 24 24"
                    fill="currentColor"
                  >
                    <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z" />
                  </svg>
                  <span>Reliable delivery</span>
                </li>
              </ul>
            </div>

            <div className={styles.infoCard}>
              <h3 className={styles.infoTitle}>Need Immediate Help?</h3>
              <p className={styles.urgentText}>
                For urgent inquiries or large orders, call us directly at{" "}
                <a href="tel:+14695551234" className={styles.phoneLink}>
                  (469) 555-1234
                </a>
              </p>
              <div className={styles.urgentActions}>
                <Button
                  text="Call Now"
                  variant="outline"
                  onClick={() => (window.location.href = "tel:+14695551234")}
                  ariaLabel="Call QS Pallets now"
                />
                <Button
                  text="Get Quote"
                  variant="secondary"
                  onClick={() => (window.location.href = "/request-a-quote")}
                  ariaLabel="Request a pallet quote"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export { Contact };
