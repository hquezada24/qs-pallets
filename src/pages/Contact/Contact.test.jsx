// Home.test.jsx
// @vitest-environment jsdom
import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Contact } from "./Contact";
import * as useContactDataModule from "../../hooks/useContactData";

// Mock the custom hook module
vi.mock("../../hooks/useContactData");

// Mock ScrollRestoration from react-router-dom
vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    ScrollRestoration: () => null,
  };
});

// Mock Button with unique test IDs based on text
vi.mock("../../components/common/Button/Index", () => ({
  Button: ({ text, onClick, disabled, loading, children }) => (
    <button
      onClick={onClick}
      disabled={disabled}
      data-testid={`button-${(text || children || "")
        .toLowerCase()
        .replace(/\s+/g, "-")}`}
      data-loading={loading}
    >
      {text || children}
    </button>
  ),
}));

// Wrapper component for Router
const renderWithRouter = (component) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};

// Create a reusable mock data fixture
const mockContactData = {
  contactInfo: {
    address: {
      street: "123 Street",
      city: "Los Angeles",
      state: "CA",
      zipCode: "00000",
      formatted: "123 Street \nLos Angeles, CA 00000",
    },
    phone: {
      display: "(000) 000-0000",
      href: "tel:+10000000000",
    },
    email: {
      display: "info@email.com",
      href: "mailto:info@email.com",
    },
    businessHours: {
      weekdays: "Monday - Friday: 7:00 AM - 6:00 PM",
      saturday: "Saturday: 8:00 AM - 4:00 PM",
      sunday: "Sunday: Closed",
    },
  },
};

describe("Contact component", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders loading state", () => {
    vi.spyOn(useContactDataModule, "useContactData").mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });

    renderWithRouter(<Contact />);

    expect(screen.getByText("Loading...")).toBeInTheDocument();
    expect(screen.queryByTestId("button-send-message")).not.toBeInTheDocument();
  });

  it("renders error state", () => {
    const errorMessage = "Failed to fetch data";
    vi.spyOn(useContactDataModule, "useContactData").mockReturnValue({
      data: null,
      loading: false,
      error: errorMessage,
    });

    renderWithRouter(<Contact />);

    expect(screen.getByText(`Error: ${errorMessage}`)).toBeInTheDocument();
    expect(screen.queryByTestId("button-send-message")).not.toBeInTheDocument();
  });

  it("renders all components when data is loaded successfully", () => {
    vi.spyOn(useContactDataModule, "useContactData").mockReturnValue({
      data: mockContactData,
      loading: false,
      error: null,
    });

    const { container } = renderWithRouter(<Contact />);

    // Test specific buttons by their unique IDs
    expect(screen.getByTestId("button-send-message")).toBeInTheDocument();
    expect(screen.getByTestId("button-call-now")).toBeInTheDocument();
    expect(screen.getByTestId("button-get-quote")).toBeInTheDocument();

    // Test content
    expect(screen.getByText(/Los Angeles/i)).toBeInTheDocument();
    expect(screen.getByText("(000) 000-0000")).toBeInTheDocument();
    expect(screen.getByText("info@email.com")).toBeInTheDocument();
    expect(
      screen.getByText("Monday - Friday: 7:00 AM - 6:00 PM")
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("handles undefined data gracefully", () => {
    vi.spyOn(useContactDataModule, "useContactData").mockReturnValue({
      data: mockContactData,
      loading: false,
      error: null,
    });

    renderWithRouter(<Contact />);

    expect(screen.getByTestId("button-send-message")).toBeInTheDocument();
  });

  it("renders successfully", () => {
    vi.spyOn(useContactDataModule, "useContactData").mockReturnValue({
      data: mockContactData,
      loading: false,
      error: null,
    });

    const { container } = renderWithRouter(<Contact />);
    expect(container.firstChild).toBeInTheDocument();
  });
});
