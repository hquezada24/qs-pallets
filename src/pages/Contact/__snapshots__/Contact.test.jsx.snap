// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact component > renders all components when data is loaded successfully 1`] = `
<div>
  <div
    class="_contactPage_9cbbc9"
  >
    <div
      class="_hero_9cbbc9"
    >
      <div
        class="_heroContent_9cbbc9"
      >
        <h1
          class="_title_9cbbc9"
        >
          Get In Touch
        </h1>
        <p
          class="_subtitle_9cbbc9"
        >
          Ready to discuss your pallet needs? We're here to help with custom solutions and reliable supply.
        </p>
      </div>
    </div>
    <div
      class="_container_9cbbc9"
    >
      <div
        class="_content_9cbbc9"
      >
        <div
          class="_formSection_9cbbc9"
        >
          <div
            class="_formHeader_9cbbc9"
          >
            <h2>
              Send Us a Message
            </h2>
            <p>
              Fill out the form below and we'll get back to you within 24 hours.
            </p>
          </div>
          <form
            class="_form_9cbbc9"
            novalidate=""
          >
            <div
              class="_formGroup_9cbbc9"
            >
              <label
                class="_label_9cbbc9"
                for="name"
              >
                Full Name *
              </label>
              <input
                class="_input_9cbbc9 "
                id="name"
                name="fullName"
                placeholder="Enter your full name"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="_formRow_9cbbc9"
            >
              <div
                class="_formGroup_9cbbc9"
              >
                <label
                  class="_label_9cbbc9"
                  for="email"
                >
                  Email Address *
                </label>
                <input
                  class="_input_9cbbc9 "
                  id="email"
                  name="email"
                  placeholder="your.email@company.com"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <div
                class="_formGroup_9cbbc9"
              >
                <label
                  class="_label_9cbbc9"
                  for="phone"
                >
                  Phone Number *
                </label>
                <input
                  class="_input_9cbbc9 "
                  id="phone"
                  name="phone"
                  placeholder="(000) 000-0000"
                  required=""
                  type="tel"
                  value=""
                />
              </div>
            </div>
            <div
              class="_formRow_9cbbc9"
            >
              <div
                class="_formGroup_9cbbc9"
              >
                <label
                  class="_label_9cbbc9"
                  for="company"
                >
                  Company Name
                </label>
                <input
                  class="_input_9cbbc9"
                  id="company"
                  name="companyName"
                  placeholder="Your Company"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="_formGroup_9cbbc9"
            >
              <label
                class="_label_9cbbc9"
                for="subject"
              >
                Subject *
              </label>
              <input
                class="_input_9cbbc9 "
                id="subject"
                name="subject"
                placeholder="Brief description of your inquiry"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="_formGroup_9cbbc9"
            >
              <label
                class="_label_9cbbc9"
                for="message"
              >
                Message *
              </label>
              <textarea
                class="_textarea_9cbbc9 "
                id="message"
                name="message"
                placeholder="Please provide details about your pallet requirements, quantities, specifications, or any questions you have."
                required=""
                rows="6"
              />
            </div>
            <div
              class="_formActions_9cbbc9"
            >
              <button
                data-loading="false"
                data-testid="button-send-message"
              >
                Send Message
              </button>
            </div>
          </form>
        </div>
        <div
          class="_infoSection_9cbbc9"
        >
          <div
            class="_infoCard_9cbbc9"
          >
            <h3
              class="_infoTitle_9cbbc9"
            >
              Contact Information
            </h3>
            <div
              class="_infoItem_9cbbc9"
            >
              <div
                class="_infoIcon_9cbbc9"
              >
                <svg
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                  />
                </svg>
              </div>
              <div
                class="_infoContent_9cbbc9"
              >
                <h4>
                  Our Location
                </h4>
                <p>
                  123 Street
                </p>
                <p>
                  Los Angeles
                  ,
                   
                  CA
                   
                  00000
                </p>
              </div>
            </div>
            <div
              class="_infoItem_9cbbc9"
            >
              <div
                class="_infoIcon_9cbbc9"
              >
                <svg
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                  />
                </svg>
              </div>
              <div
                class="_infoContent_9cbbc9"
              >
                <h4>
                  Phone
                </h4>
                <p>
                  <a
                    class="_contactLink_9cbbc9"
                    href="tel:+10000000000"
                  >
                    (000) 000-0000
                  </a>
                </p>
              </div>
            </div>
            <div
              class="_infoItem_9cbbc9"
            >
              <div
                class="_infoIcon_9cbbc9"
              >
                <svg
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </div>
              <div
                class="_infoContent_9cbbc9"
              >
                <h4>
                  Email
                </h4>
                <p>
                  <a
                    class="_contactLink_9cbbc9"
                    href="mailto:info@email.com"
                  >
                    info@email.com
                  </a>
                </p>
              </div>
            </div>
            <div
              class="_infoItem_9cbbc9"
            >
              <div
                class="_infoIcon_9cbbc9"
              >
                <svg
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                  <path
                    d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                  />
                </svg>
              </div>
              <div
                class="_infoContent_9cbbc9"
              >
                <h4>
                  Business Hours
                </h4>
                <p>
                  Monday - Friday: 7:00 AM - 6:00 PM
                </p>
                <p>
                  Saturday: 8:00 AM - 4:00 PM
                </p>
                <p>
                  Sunday: Closed
                </p>
              </div>
            </div>
          </div>
          <div
            class="_infoCard_9cbbc9"
          >
            <h3
              class="_infoTitle_9cbbc9"
            >
              Why Choose QS Pallets?
            </h3>
            <ul
              class="_benefitsList_9cbbc9"
            >
              <li
                class="_benefit_9cbbc9"
              >
                <svg
                  class="_checkIcon_9cbbc9"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
                <span>
                  Fast turnaround times
                </span>
              </li>
              <li
                class="_benefit_9cbbc9"
              >
                <svg
                  class="_checkIcon_9cbbc9"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
                <span>
                  Custom pallet solutions
                </span>
              </li>
              <li
                class="_benefit_9cbbc9"
              >
                <svg
                  class="_checkIcon_9cbbc9"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
                <span>
                  Competitive pricing
                </span>
              </li>
              <li
                class="_benefit_9cbbc9"
              >
                <svg
                  class="_checkIcon_9cbbc9"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
                <span>
                  Quality materials
                </span>
              </li>
              <li
                class="_benefit_9cbbc9"
              >
                <svg
                  class="_checkIcon_9cbbc9"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
                <span>
                  Reliable delivery
                </span>
              </li>
            </ul>
          </div>
          <div
            class="_infoCard_9cbbc9"
          >
            <h3
              class="_infoTitle_9cbbc9"
            >
              Need Immediate Help?
            </h3>
            <p
              class="_urgentText_9cbbc9"
            >
              For urgent inquiries or large orders, call us directly at
               
              <a
                class="_phoneLink_9cbbc9"
                href="tel:+14695551234"
              >
                (469) 555-1234
              </a>
            </p>
            <div
              class="_urgentActions_9cbbc9"
            >
              <button
                data-testid="button-call-now"
              >
                Call Now
              </button>
              <button
                data-testid="button-get-quote"
              >
                Get Quote
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
